Tribler/Test/test_libtorrent_download.py

```diff
@@ -70,27 +70,35 @@ def do_downloadfromfile():
         self.startTest(do_downloadfromfile)
 
     def test_downloadfromurl(self):
-        infohash = TORRENT_INFOHASH
 
         def make_screenshot():
             self.screenshot('After starting a libtorrent download from url')
+            self.file_server.stopListening()
             self.quit()
 
         def item_shown_in_list():
-            self.CallConditional(30, lambda: self.frame.librarylist.list.GetItem(infohash).original_data.ds and self.frame.librarylist.list.GetItem(
-                infohash).original_data.ds.get_current_speed(DOWNLOAD) > 0, make_screenshot, 'no download progress')
+            self.CallConditional(10, lambda: self.frame.librarylist.list.GetItem(TORRENT_FILE_INFOHASH).original_data.ds and self.frame.librarylist.list.GetItem(TORRENT_FILE_INFOHASH).original_data.ds.get_current_speed(DOWNLOAD) > 0,
+                                 make_screenshot, 'no download progress')
 
         def download_object_ready():
             self.CallConditional(10, lambda: self.frame.librarylist.list.HasItem(
-                infohash), item_shown_in_list, 'no download in librarylist')
+                TORRENT_FILE_INFOHASH), item_shown_in_list, 'no download in librarylist')
 
         def do_downloadfromurl():
             self.guiUtility.showLibrary()
-            self.frame.startDownloadFromArg(TORRENT_R, self.getDestDir())
+            self.frame.startDownloadFromArg(r'http://localhost:%s/ubuntu.torrent' % self.file_server_port,
+                                            self.getDestDir())
 
-            self.CallConditional(30, lambda: self.session.get_download(infohash), download_object_ready,
+            self.CallConditional(10, lambda: self.session.get_download(TORRENT_FILE_INFOHASH), download_object_ready,
                                  'do_downloadfromurl() failed')
 
+        # Setup file server to serve torrent file
```

Comment:
In this case not. The files to be served by the file server can differ for every test so this part we cannot extract into a separate method. Also note that the `setUpFileServer` method can be reused.